src/AssetConverter.php

```diff
@@ -0,0 +1,206 @@
+     * @param string $basePath the directory the $asset is relative to.
+     *
+     * @return string the converted asset file path, relative to $basePath.
+     */
+    public function convert(string $asset, string $basePath): string
+    {
+        $pos = strrpos($asset, '.');
+        if ($pos !== false) {
+            $srcExt = substr($asset, $pos + 1);
+            if (isset($this->commands[$srcExt])) {
+                [$ext, $command] = $this->commands[$srcExt];
+                $result = substr($asset, 0, $pos + 1).$ext;
+                if ($this->forceConvert || $this->isOutdated($basePath, $asset, $result, $srcExt, $ext)) {
+                    $this->runCommand($command, $basePath, $asset, $result);
+                }
+
+                return $result;
+            }
+        }
+
+        return $asset;
+    }
+
+    /**
+     * Checks whether asset convert result is outdated, and thus should be reconverted.
+     *
+     * @param string $basePath the directory the $asset is relative to.
+     * @param string $sourceFile the asset source file path, relative to [[$basePath]].
+     * @param string $targetFile the converted asset file path, relative to [[$basePath]].
+     * @param string $sourceExtension source asset file extension.
+     * @param string $targetExtension target asset file extension.
+     *
+     * @return bool whether asset is outdated or not.
+     */
+    protected function isOutdated(string $basePath, string $sourceFile, string $targetFile, string $sourceExtension, string $targetExtension): bool
+    {
+        $resultModificationTime = @filemtime("$basePath/$targetFile");
+        if ($resultModificationTime === false || $resultModificationTime === null) {
+            return true;
+        }
+
+        if ($resultModificationTime < @filemtime("$basePath/$sourceFile")) {
+            return true;
+        }
+
+        if ($this->isOutdatedCallback === null) {
+            return false;
+        }
+
+        return call_user_func($this->isOutdatedCallback, $basePath, $sourceFile, $targetFile, $sourceExtension, $targetExtension);
+    }
+
+    /**
+     * Runs a command to convert asset files.
+     *
+     * @param string $command the command to run. If prefixed with an `@` it will be treated as a [path alias](guide:concept-aliases).
+     * @param string $basePath asset base path and command working directory
+     * @param string $asset the name of the asset file
+     * @param string $result the name of the file to be generated by the converter command
+     *
+     * @throws \Exception when the command fails and YII_DEBUG is true.
+     * In production mode the error will be logged.
+     *
+     * @return bool true on success, false on failure. Failures will be logged.
+     */
+    protected function runCommand(string $command, string $basePath, string $asset, string $result): bool
+    {
+        $command = $this->aliases->get($command);
```

Comment:
Why it needs?

Reply:
So it is in the code since Yii2, I must verify it.